compiler/core/src/utils/monads.re

```diff
@@ -0,0 +1,7 @@
+let bindOption = (f: 'a => option('b), value: option('a)): option('b) =>
```

Comment:
Let's just call this file `monad.re` to be slightly more consistent with haskell's naming (theirs is `import Control.Monad`)